layers/writer.go

```diff
@@ -35,8 +35,12 @@ func (lw *layerWriter) Digest() string {
 }
 
 func tarWriter(lw *layerWriter) *archive.NormalizingTarWriter {
+	var tw *archive.NormalizingTarWriter
 	if runtime.GOOS == "windows" {
-		return archive.NewNormalizingTarWriter(layer.NewWindowsWriter(lw))
+		tw = archive.NewNormalizingTarWriter(layer.NewWindowsWriter(lw))
+	} else {
+		tw = archive.NewNormalizingTarWriter(tar.NewWriter(lw))
```

Comment:
Are you comfortable with the lack of symmetry here?  That there's a custom windows writer, but not a custom *nix writer?

Reply:
I'll admit it does bother me a little. However, windows layers needs special entries and prefixes and linux layers don't need anything special. If we moved some of generic layer helpers to `imgutil` it would look symmetric to the consumers. This  would be a nice change, but not one I am willing to make right now.